src/common/src/util/ordered/serde.rs

```diff
@@ -271,4 +298,223 @@ mod tests {
             OrderedRow::new(row3, &order_types)
         );
     }
+
+    #[test]
+    fn test_deserialize_with_column_indices() {
+        let order_types = vec![OrderType::Descending, OrderType::Ascending];
+        let serializer = OrderedRowSerializer::new(order_types.clone());
+        let schema = vec![DataType::Varchar, DataType::Int16];
+        let row1 = Row(vec![Some(Utf8("abc".to_string())), Some(Int16(5))]);
+        let row2 = Row(vec![Some(Utf8("abd".to_string())), Some(Int16(5))]);
+        let row3 = Row(vec![Some(Utf8("abc".to_string())), Some(Int16(6))]);
+        let rows = vec![row1, row2, row3];
+        let deserializer = OrderedRowDeserializer::new(schema, order_types);
+        let mut array = vec![];
+        for row in &rows {
+            let mut row_bytes = vec![];
+            serializer.serialize(row, &mut row_bytes);
+            array.push(row_bytes);
+        }
+
+        let row_len = array[0].len();
+        let row_0_idx_0_len = deserializer
+            .deserialize_prefix_len_with_column_indices(&array[0], vec![0])
+            .unwrap();
+        assert_eq!(11, row_0_idx_0_len);
```

Comment:
I suggest creating a deserializer to deserialize the prefix slice to see whether the result is the same as the original row instead of assert_eq on a manually calculated length.

Reply:
fix this case through deserializer prefix_slice and check the datum